src/App/App.js

```diff
@@ -5,38 +5,41 @@ import Header from '../Header/Header';
 import Modal from '../Modal/Modal';
 // import RabbitTrail from '../RabbitTrail/RabbitTrail'
 import CardContainer from '../CardContainer/CardContainer';
+import {getAllMovies, getMovie } from '../util.js';
 import movieData from '../movieData';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movieData: movieData.movies,
+      movieData: [],
       movieID: 0,
       foundMovie: null,
       modalShowing: false
     }
   }
 
-  // funciton here?
-  // get click event
-  // happens in card container
-
-
-  // function that shows movie will be defined in app
-  // pass this function into card container to grab the event.target
-  // return that object (single movie) here
-  // pass that object into modal for further destructuring
+  componentDidMount() {
+    getAllMovies()
+    .then(movies => {
+      console.log(movies);
+      this.setState({movieData: movies.movies})
+    })
+    .catch(error => console.log(error))
+  }
 
-  getMovie = (event) => {
+  setFoundMovie = (event) => {
     const movieID = event.target.closest("article").id;
-    const foundMovie = this.getAllMovieData(movieID);
-
-    this.setState({
-      movieID: movieID,
-      modalShowing: true,
-      foundMovie: foundMovie
-    });
+    getMovie(movieID)
+    .then(movie => {
+      console.log(movie.movie);
```

Comment:
console.log :)

Also, I love that it's so easy to toss this data into state from within a promise.